spec/generator/gen-named-callout.spec.js

```diff
@@ -0,0 +1,107 @@
+import {toOnigurumaAst} from '../../dist/index.js';
+import {generate} from '../../dist/generator/generate.js';
+
+describe('generate: Named Callout', () => {
+  function gen(pattern) {
+    return generate(toOnigurumaAst(pattern)).pattern;
+  }
+
+  it('should support all named callout names', () => {
+    const cases = [
+      ['(*FAIL)'],
+      ['(*MISMATCH)'],
+      ['(*SKIP)'],
+      ['(*ERROR)'],
+      ['(*MAX{1})'],
+      ['(*COUNT)'],
+      ['(*TOTAL_COUNT)'],
+      ['(*CMP{5,>=,4})'],
+      // ['(*foo)'], // Error: Invalid callout name "foo"
+    ];
+    for (const [input, expected] of cases) {
+      expect(gen(input)).toBe(expected ?? input);
+    }
```

Comment:
For cases like this where there are no changes to the input, better to change the `cases` array from an array of arrays to an array of strings, and do:

```js
for (const input of cases) {
  expect(gen(input)).toBe(input);
}
```

I should also make this change for one of the current functions in `gen-quantifier.spec.js`.

Reply:
https://github.com/slevithan/oniguruma-parser/commit/f09e46046360563df0a032b644b4db88a281d2fd